packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -123,12 +137,59 @@ class ImageStreamListener {
 /// same stack frame as the call to [ImageStream.addListener]).
 typedef ImageListener = void Function(ImageInfo image, bool synchronousCall);
 
+/// Signature for listening to [ImageChunkEvent] events.
+///
+/// Used in [ImageStreamListener].
+typedef ImageChunkListener = void Function(ImageChunkEvent event);
+
 /// Signature for reporting errors when resolving images.
 ///
 /// Used in [ImageStreamListener], as well as by [ImageCache.putIfAbsent] and
 /// [precacheImage], to report errors.
 typedef ImageErrorListener = void Function(dynamic exception, StackTrace stackTrace);
 
+/// An immutable notification of image bytes that have been incrementally loaded.
+///
+/// Chunk events represent progress notifications while an image is being
+/// loaded (e.g. from disk or over the network).
+///
+/// See also:
+///
+///  * [ImageChunkListener], the means by which callers get notified of
+///    these events.
+@immutable
+class ImageChunkEvent extends Diagnosticable {
+  /// Creates a new chunk event.
+  const ImageChunkEvent(this.cumulativeBytesLoaded, this.expectedTotalBytes)
+      : assert(cumulativeBytesLoaded >= 0),
+        assert(expectedTotalBytes >= -1);
+
+  /// The number of bytes that have been received across the wire thus far.
+  final int cumulativeBytesLoaded;
+
+  /// The expected number of bytes that need to be received to finish loading
+  /// the image.
+  ///
+  /// This value is not necessarily equal to the expected _size_ of the image
+  /// in bytes, as the bytes required to load the image may be compressed.
+  ///
+  /// This value will be -1 if the number is not known in advance. In this way,
```

Comment:
Wouldn't `null` be a better choice over a magic number to represent that this information is unavailable?

Reply:
Done